scipy/integrate/_quadrature.py

```diff
@@ -669,7 +670,7 @@ def simpson(y, x=None, dx=1.0, axis=-1, even=None):
         slice_all = (slice(None),) * nd
 
         # default is 'simpson'
-        even = even if even is not None else "simpson"
+        even = even if even is not _NoValue else "simpson"
```

Comment:
This should also check if `even` is not `None`, e.g.:
```python
even = even if even is not in (_NoValue, None) else "simpson"
```
or alternatively, undo the change in this line and add this right below the deprecation warning block:
```python
else:
    even = None
```